Modules/posixmodule.c

```diff
@@ -6999,10 +6999,29 @@ posix_getgrouplist(PyObject *self, PyObject *args)
     if (groups == NULL)
         return PyErr_NoMemory();
 
+#ifdef __APPLE__
+    while (getgrouplist(user, basegid, groups, &ngroups)) {
+        /* On Darwin 19.3 (macOS 10.15 Catalina), getgrouplist() returns a
+           non-zero value without setting errno if the group list is too small.
+           Double the list size and call it again in this case. */
+        PyMem_Free(groups);
+
+        if (ngroups > INT_MAX / 2) {
```

Comment:
Just question:
I 've understood the situation.

Is there any reason not to try with INT_MAX for the last trying?

```C

if (ngroups == INT_MAX) {
   return PyErr_NoMemory();
} else if (ngroups > INT_MAX / 2) {
    ngroups = INT_MAX;
} else {
    ngroups *= 2;
}
```

Reply:
If `(ngroups > INT_MAX / 2)` is false, `ngroups *= 2` would overflow: ngroups type is `int`.

```
} else if (ngroups > INT_MAX / 2) {
    ngroups = INT_MAX;
```

I don't think that it's worth it to bother with attempting that. I don't think that any user has more than 1073741823 groups. Such user is going to get larger issues than that.

Reply:
Okay got it :) Sounds reasonable.